main.c

```diff
@@ -687,30 +689,23 @@ int main(void)
 	gpio_put(PICO_DEFAULT_LED_PIN, 1);
 
 	struct image_header *hdr = (struct image_header *)(XIP_BASE + IMAGE_HEADER_OFFSET);
-
-	gpio_set_function(UART_TX_PIN, GPIO_FUNC_UART);
+    
+  gpio_set_function(UART_TX_PIN, GPIO_FUNC_UART);
 	gpio_set_function(UART_RX_PIN, GPIO_FUNC_UART);
-	uart_init(uart0, UART_BAUD);
-	uart_set_hw_flow(uart0, false, false);
-	
-	sleep_ms(10);
-
-	    while (uart_is_readable(uart0)){uart_getc(uart0);}; //clear Buffer
+	uart_init(UART_ID, UART_BAUD);
+	uart_set_hw_flow(UART_ID, false, false);
     
-	//Wait loop
-    bool no_rec = true;
-
-    uart_puts(uart0, "Boot\r\n");  //Header
+	sleep_ms(10);  // waiting for Serial line to fully initialize
+ 
+    bool nothing_received = true;
+    uart_puts(UART_ID, BL_IDENTIFIER_STR);  //Header
 
-	for(int i = 0; i < 10000; i++){
-       busy_wait_us(100);
-       if (uart_is_readable(uart0)){     
-           no_rec = false;
-		   i = 11000;
-		}
+	if(uart_is_readable_within_us(UART_ID,1000000)) {
```

Comment:
Magic number, kindly create a define for that in the config header file

Reply:
done